otter/convergence/steps.py

```diff
@@ -93,12 +93,20 @@ class DeleteServer(object):
 
     def as_effect(self):
         """Produce a :obj:`Effect` to delete a server."""
-        return service_request(
+        eff = service_request(
             ServiceType.CLOUD_SERVERS,
             'DELETE',
             append_segments('servers', self.server_id),
             success_pred=has_code(204))
 
+        def report_success(result):
```

Comment:
Perhaps, eventually, but not in this PR. It is clear that we don't actually know what they're going to end up looking like; most steps need very distinct things. I want to know the actual requirements before I refactor it into something we can't actually use.